packages/flame/lib/src/components/components_notifier.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+/// A [ChangeNotifier] that notifies its listeners when a [Component] is
+/// added or removed, or updated. The meaning of an updated component
+/// will vary depending on the component implementation, this is something
+/// defined and executed by the component itself.
+///
+/// For example, in a Player component, that holds a health variable
+/// may notify changes when that variable has changed.
+class ComponentsNotifier<T extends Component> extends ChangeNotifier {
+  ComponentsNotifier(List<T> initial) : _components = initial;
+
+  final List<T> _components;
+
+  /// The list of components.
+  List<T> get components => _components;
```

Comment:
This could just be final, the getter doesn't protect anything.

Reply:
I was going to put an `UnmodifiableListView` wrapping this! Good thing that you commented, I've added it now